gtas-parent/gtas-commons/src/main/java/gov/gtas/services/matcher/quickmatch/MatchingContext.java

```
@@ -406,29 +407,24 @@ private DerogResponse matchTraveler(HashMap<String, String> traveler) {
         return new DerogResponse(traveler.get("gtasId"), derogHits);
     }
 
-    /**
-     * 
-     * @param travelerDate
-     * @param deragDate
-     * @return
-     * @throws ParseException
-     */
+	/**
+	 * 
+	 * @param travelerDate
+	 * @param deragDate
+	 * @return
+	 * @throws ParseException
+	 */
 	private int calculateYearOffset(String travelerDate, String deragDate) {
 
```

Due to the performance needs of quick matcher I think two things could be considered here.

1. Move the formatter to the contructor as a class level variable. 
2. Use Java 8 libraries as they are significantly faster. The below code does the same thing but should be at least 2.5x faster.

	private int calculateYearOffset(String travelerDate, String deragDate) {
		DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd", Locale.ROOT);
        try {
            LocalDate travLocalDate = LocalDate.parse(travelerDate, dateTimeFormatter);
            LocalDate derogLocalDate = LocalDate.parse(deragDate, dateTimeFormatter);
            return Math.abs(travLocalDate.getYear() - derogLocalDate.getYear());
        } catch (Exception e) {
            logger.error("Failure to parse date");
        }
        return -1;
	}

---

Also, I think the"isSameDobYearMonthDay" method could be reduced to 

	private boolean isSameDOBYearMonthDay(String travelerDate, String deragDate)  {
			return travelerDate.equalsIgnoreCase(deragDate);
	}

---

That would fail if the two dates are formatted differently.

---

I do not think there is an expectation to get or make watchlist items with more than year-month-date, We control how the traveler bithdate is formatted within the class. Hopefully that is a case that while currently can be fed into the matcher it would not occur. Perhaps we can throw an exception if the date is not formatted as we expect and put a requirement on the class to get dates formatted in yyyy-MM-dd.

Depending on the performance gains from using dateTimeFormatter this might be an acceptable requirement to put on the class - it would also open up the string comparison for exact date match which is likely faster than instantiating objects to immediately compare and discard them.